lib/metric-config-parser/metric_config_parser/tests/test_experiment.py

```diff
@@ -267,6 +267,46 @@ def test_preenrollment(self, experiments, config_collection):
         assert len(days28_metrics) == 1
         assert days28_metrics[0].metric.name == "spam"
 
+    def test_segment_resolution(self, experiments, config_collection):
+        spec = AnalysisSpec()
+
+        cfg = spec.resolve(experiments[9], config_collection)
+        segments = cfg.experiment.segments
+
+        assert any(
+            segment.name == "regular_users_v3" and segment.data_source.name == "my_cool_data_source"
+            for segment in segments
+        )
+
+    def test_segment_definition_and_experiment_segment(self, experiments, config_collection):
+        config_str = dedent(
+            """
+            [experiment]
+            segments = ["test_segment"]
+
+            [segments.test_segment]
+            Data_Source = "test_data_source"
+            Select_Expression = "{{agg_any('1')}}"
+
+            [segments.data_sources.test_data_source]
+            from_expression = "(SELECT 1 WHERE submission_date BETWEEN {{experiment.start_date_str}} AND {{experiment.last_enrollment_date_str}})"
+            """  # noqa
+        )
+        spec = AnalysisSpec.from_dict(toml.loads(config_str))
+        cfg = spec.resolve(experiments[9], config_collection)
+
+        custom_segment = [seg for seg in cfg.experiment.segments if seg.name == "test_segment"]
```

Comment:
Like above, I'd prefer that we assert the length of the segments, and then retrieve them individually and make further assertions. I'd also suggest the same regarding the data sources `from_expression` resolution.

I think asserting that `len(custom_segment) == 1` is still valid to ensure we're deduping properly, but it might be better to put that in a separate test where we add the `experiments[9]` segment to the list of custom experiment config segments.

Reply:
Just needs an assertion on the total length of segments:
```suggestion
        assert len(cfg.experiment.segments) == 2

        custom_segment = [seg for seg in cfg.experiment.segments if seg.name == "test_segment"]
```